adlfs/spec.py

```diff
@@ -398,7 +398,11 @@ def _strip_protocol(cls, path: str):
 
         STORE_SUFFIX = ".dfs.core.windows.net"
         logger.debug(f"_strip_protocol for {path}")
-        if not path.startswith(("abfs://", "az://", "abfss://")):
+        if isinstance(cls.protocol, str):
+            protocol_startswith = (f"{cls.protocol}://",)
```

Comment:
(optional) add comment that this covers the case that a protocol was supplied to the instance, probably by the user.